server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java

```diff
@@ -72,18 +73,52 @@ protected BoolQuery createBoolQueryForProperties(List<Query> must, List<Query> s
         return builder.build();
     }
 
+    public ResponseEntity<List<String>> getAutocompleteSuggestions(String input) throws IOException {
+        Map<String, FieldSuggester> map = new HashMap<>();
+        map.put(APIConstants.TITLE_SUGGEST_NAME, FieldSuggester.of(fs -> fs
+                .completion(cs -> cs.skipDuplicates(true)
+                        .size(10)
+                        .fuzzy(SuggestFuzziness.of(sf -> sf.fuzziness("1").minLength(2)))
+                        .field(APIConstants.TITLE_SUGGEST))
+        ));
+        Suggester suggester = Suggester.of(s -> s
+                .suggesters(map)
+                .text(input)
+        );
+        SearchRequest searchRequest = SearchRequest.of(s -> {
+            s.index(indexName)
+                    // can add more fields to be returned
+                    .source(SourceConfig.of(sc -> sc.filter(f -> f.includes(List.of(APIConstants.TITLE_FIELD)))))
```

Comment:
It will be good if this is a parameter in the yaml so that we do not need to change code if more field is needed

Then you do not need to create constants, but if you want to type check the value, you can use Enum